packages/neo-one-client/src/user/keystore/LedgerKeyStore.ts

```diff
@@ -0,0 +1,232 @@
+import { LedgerHandler } from './LedgerHandler';
+
+export interface Ledger {
+  readonly accKey: number;
+  readonly account: UserAccount;
+}
+
+export type Ledgers = { readonly [Network in string]?: { readonly [Address in string]?: Ledger } };
+
+export interface LedgerProvider {
+  readonly network: NetworkType;
+  readonly getAccount: (address: AddressString, monitor?: Monitor) => Promise<Account>;
+}
+
+const flattenLedgers = (ledgers: Ledgers) =>
+  _.flatten(
+    Object.values(ledgers)
+      .filter(utils.notNull)
+      .map((networkLedgers) => Object.values(networkLedgers)),
+  ).filter(utils.notNull);
+
+export class LedgerKeyStore {
+  public get ledgers(): Ledgers {
+    return this.ledgersInternal$.getValue();
+  }
+
+  public get currentAccount(): UserAccount | undefined {
+    return this.currentAccountInternal$.getValue();
+  }
+  public readonly type: string;
+  public readonly byteLimit: number;
+  public readonly currentAccount$: Observable<UserAccount | undefined>;
+  public readonly accounts$: Observable<ReadonlyArray<UserAccount>>;
+  public readonly ledgers$: Observable<ReadonlyArray<Ledger>>;
+  public readonly initPromise: Promise<LedgerHandler>;
+
+  private readonly provider: LedgerProvider;
+  private readonly currentAccountInternal$: BehaviorSubject<UserAccount | undefined>;
+  private readonly accountsInternal$: BehaviorSubject<ReadonlyArray<UserAccount>>;
+  private readonly ledgersInternal$: BehaviorSubject<Ledgers>;
+
+  public constructor(provider: LedgerProvider) {
+    this.type = LedgerHandler.type;
+    this.byteLimit = LedgerHandler.byteLimit;
+    this.provider = provider;
+
+    this.ledgersInternal$ = new BehaviorSubject<Ledgers>({});
+    this.ledgers$ = this.ledgersInternal$.pipe(
+      distinctUntilChanged((a, b) => _.isEqual(a, b)),
+      map(flattenLedgers),
+    );
+
+    this.accountsInternal$ = new BehaviorSubject([] as ReadonlyArray<UserAccount>);
+    this.ledgers$.pipe(map((ledgers) => ledgers.map(({ account }) => account))).subscribe(this.accountsInternal$);
+    this.accounts$ = this.accountsInternal$;
+
+    this.currentAccountInternal$ = new BehaviorSubject(undefined as UserAccount | undefined);
+    this.currentAccount$ = this.currentAccountInternal$.pipe(distinctUntilChanged());
+
+    this.initPromise = this.init();
+  }
+
+  public getCurrentAccount(): UserAccount | undefined {
+    return this.currentAccountInternal$.getValue();
+  }
+
+  public getAccounts(): ReadonlyArray<UserAccount> {
+    return this.accountsInternal$.getValue();
+  }
+  public async sign({
+    account,
+    message,
+    monitor,
+  }: {
+    readonly account: UserAccountID;
+    readonly message: string;
+    readonly monitor?: Monitor;
+  }): Promise<Witness> {
+    const handler = await this.initPromise;
+
+    return this.capture(
+      async () => {
+        const ledger = this.getLedger(account);
+        const response = await handler.sign({ message, account: ledger.accKey });
```

Comment:
I would just make the `LedgerHandler` work with `Buffer`s exclusively - it's not a public API so no need to be converting back and forth between strings.